tap_klaviyo/streams.py

```diff
@@ -75,12 +82,94 @@ def post_process(
         context: dict | None = None,  # noqa: ARG002
     ) -> dict | None:
         row["updated_at"] = row["attributes"]["updated_at"]
+        row["tags"] = [self.included_map[tag["id"]]
+                       for tag in row.get("relationships", {}).get("tags", {}).get("data", [])]
+        row["campaign_messages"] = [self.included_map[campaign_message["id"]]
+                                    for campaign_message in
+                                    row.get("relationships", {}).get("campaign-messages", {}).get("data", [])]
         return row
 
     @property
     def is_sorted(self) -> bool:
         return True
 
+    def get_child_context(self, record: dict, context: t.Optional[dict]) -> dict:
+        """Return a context dictionary for child streams."""
+        return {
+            "campaign_id": record["id"],
+            "campaign_status": record["attributes"]["status"],
+        }
+
+    def parse_response(self, response: requests.Response) -> t.Iterable[dict]:
+        self.process_included(response)
```

Comment:
The included fields are returned outside of data rows, so we need to join them during the process.